File: News_Service_JDBC/src/main/java/com/dunice/projectNews/models/Tag.java
package com.dunice.projectNews.models;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.ManyToMany;
import jakarta.persistence.Table;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import java.util.Set;

@Setter
@Getter
@Entity
@Table(name = "tags", schema = "news_schema")
@NoArgsConstructor
public class Tag {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "tag_id", nullable = false)
    private Long tagId;

    @ManyToMany(mappedBy = "tags")
    private Set<News> news;

    @Column(name = "title", nullable = false)
    private String title;
}

File: News_Service_Kotlin/src/main/kotlin/com/dunice/advancedServiceKotlin/models/Tag.kt
package com.dunice.advancedServiceKotlin.models

import News
import jakarta.persistence.*

@Entity
@Table(name = "tags", schema = "news_schema")
 class Tag(

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "tag_id", nullable = false)
    var tagId: Long=0L,

    @ManyToMany(mappedBy = "tags")
    var news: Set<News> = HashSet(),

    @Column(name = "title", nullable = false)
    var title: String="",
)
